### PR TITLE
add better sync of calendar

### PR DESCRIPTION
**Pull request by inkhey**
:warning:  rebase from maintenance branch !

related to #5263 

## Checkpoints

**For developers**

- [ ] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [ ] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [ ] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [ ] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [ ] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [ ] The code is clear enough
- [ ] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [ ] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [ ] Manual, quality tests have been done

Todo:
- [ ] set origin to maintenance branche
- [ ] add few more automated tests:
   - [ ] cleanup api case (agenda délétion)
   - [ ]  disabling workspace agenda link when user ils deleted/disabled

### CODE REVIEW

File: backend/tracim_backend/applications/agenda/lib.py [hunk 1]
```diff
@@ -65,6 +67,12 @@
 """
 
 
+class AgendaCreationState(str, Enum):
```
Comment (grignards) [inline backend/tracim_backend/applications/agenda/lib.py:70]: Could you add a comment to explain the differences between `CREATED` and `EXISTING`?

**Review by grignards — APPROVED** [review]

**Comment by grignards** [comment]
@inkhey don't forget that the target branch must be `maintenance/4.0`

**Comment by grignards** [comment]
Closing as the target branch should be `maintenance/4.1`. New PR: https://github.com/tracim/tracim/pull/5323

File: backend/tracim_backend/applications/agenda/lib.py (hunk 1 above)
Comment (inkhey) [reply backend/tracim_backend/applications/agenda/lib.py:70]: done.

**Pull request closed without merging**